### PR TITLE
block building or fetching should not occur if sync is in progress

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

**Review by scottmillner — DISMISSED** [review]

File: crates/consensus/authority/src/block_builder.rs [hunk 1]
```diff
@@ -39,6 +39,13 @@ where
         ConfigureEvmEnv + Clone + Unpin + Send + Sync + 'static + reth_node_api::ConfigureEvm,
 {
     pub(crate) async fn try_build_block(&mut self) {
+        // ensure the node is not syncing
+        if is_active_sync_in_progress(&self.network_handle) {
+            warn!(target: "consensus::authority", "Node is still syncing, block builder task is awaiting fully synced status ...");
+            tokio::time::sleep(Duration::from_secs(2)).await;
```
Comment (scottmillner) [inline crates/consensus/authority/src/block_builder.rs:45]: just curious on why 2 seconds?

Comment (BatiGencho) [reply crates/consensus/authority/src/block_builder.rs:45]: its just a value to introduce some relaxation within the loop not to keep the runtime busy.

File: crates/consensus/authority/src/pbft_task.rs [hunk 2]
```diff
@@ -53,6 +58,10 @@ pub struct PbftTask<Client, ToFrostMan: ToFrostManager, NetworkClient> {
     secret_key: secp256k1::SecretKey,
     /// config
     config: FrostConfig,
+    /// network client
+    network_client: NetworkClient,
```
Comment (arminsabouri) [inline crates/consensus/authority/src/pbft_task.rs:62]: Do we need network client?

**Review by arminsabouri — DISMISSED** [review]
Nicely done!

File: crates/consensus/authority/src/pbft_task.rs (hunk 2 above)
Comment (BatiGencho) [reply crates/consensus/authority/src/pbft_task.rs:62]: yes, the pbft state machine needs it.

**Review by scottmillner — APPROVED** [review]

**Review by arminsabouri — APPROVED** [review]

**Comment by arminsabouri** [comment]
./test_runner passing locally. Merging

**Pull request merged**